spring-cloud-aws-cloudmap/src/main/java/org/springframework/cloud/aws/cloudmap/CloudMapRegistryService.java

```diff
@@ -0,0 +1,370 @@
+			}
+			catch (MaxRetryExceededException e) {
+				log.error("Maximum number of retry exceeded for registering instance with {} for {}", nameSpace,
+						service, e);
+			}
+		}
+		else {
+			log.info("Service registration skipped");
+		}
+
+		return null;
+	}
+
+	/**
+	 * Create Cloudmap namespace.
+	 * @param properties cloudmap properties
+	 * @param vpcId VPC ID
+	 * @return NamespaceID
+	 * @throws CreateNameSpaceException thrown in case of runtime exception
+	 */
+	private String createNameSpace(CloudMapRegistryProperties properties, String vpcId)
+			throws CreateNameSpaceException {
+		final String nameSpace = properties.getServiceNameSpace();
+		try {
+			// Create namespace
+			final String operationId = serviceDiscovery.createPrivateDnsNamespace(new CreatePrivateDnsNamespaceRequest()
+					.withName(nameSpace).withVpc(vpcId).withDescription(properties.getDescription())).getOperationId();
+
+			// Wait till completion
+			pollForCompletion(operationId);
+
+			return getNameSpaceId(nameSpace);
+		}
+		catch (NamespaceAlreadyExistsException e) {
+			return getNameSpaceId(nameSpace);
+		}
+		catch (InvalidInputException | ResourceLimitExceededException | DuplicateRequestException e) {
+			log.error("Error while registering with cloudmap {} with error {}", nameSpace, e.getMessage(), e);
+			throw new CreateNameSpaceException(e);
+		}
+		catch (InterruptedException e) {
+			log.error("Error while polling for status update {} with error {}", nameSpace, e.getMessage(), e);
+			throw new CreateNameSpaceException(e);
+		}
+		catch (MaxRetryExceededException e) {
+			log.error("Maximum number of retry exceeded for namespace {}", nameSpace, e);
+			throw new CreateNameSpaceException(e);
+		}
+	}
+
+	/**
+	 * Create service.
+	 * @param nameSpaceId CloudMap Namespace ID
+	 * @return Service ID
+	 * @throws CreateServiceException thrown in case of runtime exception
+	 */
+	private String createService(String nameSpaceId) throws CreateServiceException {
+		final String nameSpace = properties.getServiceNameSpace();
+		final String service = properties.getService();
+
+		try {
+			CreateServiceRequest serviceRequest = new CreateServiceRequest().withName(service)
+					.withNamespaceId(nameSpaceId).withDnsConfig(
+							new DnsConfig().withDnsRecords(new DnsRecord().withType(RecordType.A).withTTL(300L)));
+
+			if (!StringUtils.isNullOrEmpty(properties.getHealthCheckResourcePath())) {
+				HealthCheckConfig healthCheckConfig = new HealthCheckConfig();
+				healthCheckConfig.setType(String.valueOf(("https").equalsIgnoreCase(properties.getHealthCheckProtocol())
+						? HealthCheckType.HTTPS : HealthCheckType.HTTP));
+				healthCheckConfig.setResourcePath(properties.getHealthCheckResourcePath());
+				healthCheckConfig.setFailureThreshold(
+						(properties.getHealthCheckThreshold() == null) ? 5 : properties.getHealthCheckThreshold());
+				serviceRequest.withHealthCheckConfig(healthCheckConfig);
+			}
```

Comment:
Health checks are supported only for `public` or `HTTP` namespaces. It doesn't work for `private` namespace that are used here.

https://docs.aws.amazon.com/cloud-map/latest/api/API_CreateService.html

Reply:
Good catch. I have removed these additional methods and code